app/navigators/EstablishersCompanyDirectorNavigator.scala

```diff
@@ -19,100 +19,132 @@ package navigators
-      case ConfirmDeleteDirectorId(establisherIndex) =>
-        NavigateTo.dontSave(controllers.register.establishers.company.routes.AddCompanyDirectorsController.onPageLoad(NormalMode, None, establisherIndex))
-      case CheckYourAnswersId(establisherIndex, directorIndex) =>
-        NavigateTo.save(controllers.register.establishers.company.routes.AddCompanyDirectorsController.onPageLoad(NormalMode, None, establisherIndex))
-      case _ => None
+        checkYourAnswers(establisherIndex, directorIndex, mode, srn)
+      case DirectorAddressYearsId(establisherIndex, directorIndex) =>
+        addressYearsRoutes(establisherIndex, directorIndex, mode, srn)(from.userAnswers)
+      case _ => commonRoutes(from, mode, srn)
     }
-  }
 
-  override protected def editRouteMap(from: NavigateFrom): Option[NavigateTo] = {
+  protected def editRoutes(from: NavigateFrom, mode: Mode, srn: Option[String]): Option[NavigateTo] =
     from.id match {
       case DirectorDetailsId(establisherIndex, directorIndex) =>
-        checkYourAnswers(establisherIndex, directorIndex)(from.userAnswers)
-      case DirectorNinoId(establisherIndex, directorIndex) =>
-        checkYourAnswers(establisherIndex, directorIndex)(from.userAnswers)
+        exitMiniJourney(establisherIndex, directorIndex, mode, srn)
+      case DirectorNinoId(establisherIndex, directorIndex)  =>
+        exitMiniJourney(establisherIndex, directorIndex, mode, srn)
       case DirectorUniqueTaxReferenceId(establisherIndex, directorIndex) =>
-        checkYourAnswers(establisherIndex, directorIndex)(from.userAnswers)
-      case DirectorAddressPostcodeLookupId(establisherIndex, directorIndex) =>
-        NavigateTo.save(routes.DirectorAddressListController.onPageLoad(CheckMode, establisherIndex, directorIndex, None))
-      case DirectorAddressListId(establisherIndex, directorIndex) =>
-        NavigateTo.save(routes.DirectorAddressController.onPageLoad(CheckMode, establisherIndex, directorIndex, None))
+        exitMiniJourney(establisherIndex, directorIndex, mode, srn)
       case DirectorAddressId(establisherIndex, directorIndex) =>
-        checkYourAnswers(establisherIndex, directorIndex)(from.userAnswers)
+        exitMiniJourney(establisherIndex, directorIndex, mode, srn)
+      case DirectorPreviousAddressId(establisherIndex, directorIndex) =>
+        exitMiniJourney(establisherIndex, directorIndex, mode, srn)
+      case DirectorContactDetailsId(establisherIndex, directorIndex) =>
+        exitMiniJourney(establisherIndex, directorIndex, mode, srn)
       case DirectorAddressYearsId(establisherIndex, directorIndex) =>
-        addressYearsEditRoutes(establisherIndex, directorIndex)(from.userAnswers)
+        addressYearsEditRoutes(establisherIndex, directorIndex, mode, srn)(from.userAnswers)
+      case _ => commonRoutes(from, mode, srn)
+    }
+
+  protected def commonRoutes(from: NavigateFrom, mode: Mode, srn: Option[String]): Option[NavigateTo] =
```

Comment:
Cyclomatic complexity warning here. Either reduce complexity or switch off this check with scalastyle:off

Reply:
unable to find it